inference_rules.adoc

```diff
@@ -821,6 +821,30 @@ The DLRMv2 MLPerf inference code has an option to aggregate multiple consecutive
 2. drawn uniformly from interval [`--samples-to-aggregate-min`, `--samples-to-aggregate-max`]	
 3. drawn from a custom distribution, with its quantile (inverse of CDP) specified in `--samples-to-aggregate-quantile-file=./tools/dist_quantile.txt`.
 
+=== LLM Benchmarks
+
+
+LLM benchmarks contain an auto-regressive generation loop. This loop’s execution is controlled by a decoding algorithm. The following rules discuss the decoding algorithm in the light of MLPerf Inference rules for the Closed submissions:
+
+* Decoding algorithm should follow the implementation available on huggingface: https://huggingface.co/blog/how-to-generate#beam-search 
+  ** We are using beam=4 for this algorithm’s setting.
+  ** The output length determines the runtime of the generation. So, the algorithm in a submission should follow the same settings as the reference model and the algorithm should not change the output length.
+* Summary of the allowed decoding algorithm:
+  ** For the LLM benchmarks, each query contains a prompt. Upon receiving the query, the SUT runs this query in two stages: 
+    1. Run the forward pass of the model. As part of this, populate the KV-cache corresponding to this request.
+    2. Start the decoding loop using the beam search algorithm, using 4 beams.
+    3. Loop ends when the max tokens for the benchmark is reached or EOS token is detected as part of beam search.
```

Comment:
Please describe early_stopping=True from HF::
Stop once:
•	Number of generated tokens in all beams reaches max_new_tokens=128 or
•	We found num_beams(=4) candidates.